sound/soc/sof/intel/hda-bus.c

```diff
@@ -74,14 +74,34 @@ static const struct hdac_io_ops io_ops = {
 	.dma_free_pages = sof_hda_dma_free_pages,
 };
 
+static u64 bus_idx_bits;
+
+static int get_free_index(void)
+{
+	int i = 0;
+
+	while ((bus_idx_bits >> i) & 0x1) {
+		if (i++ == 63)
+			return -EINVAL;
```

Comment:
you are not testing the return value, so this error case will not really do anything.

Reply:
As I summarize, my new thinking is that we can just change to "bus->idx = 0" or even don't touch it(default 0) at sof_hda_bus_init(), @plbossart what do you think about that?